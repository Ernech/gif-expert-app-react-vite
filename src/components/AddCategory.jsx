import React, { useState } from 'react'
import PropTypes from 'prop-types'
export const AddCategory = ({onNewCategory}) => {

    
    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) =>{
        setInputValue(target.value)
    }

    const submitForm = (event) =>{
        event.preventDefault()
        const newInputVale=inputValue.trim()
        if(newInputVale.length<=1) return
       onNewCategory( newInputVale )
       setInputValue('')

    }

  return (
    <form onSubmit={submitForm} aria-label='form'>
        <input type='text' placeholder='Buscar gifs' value={inputValue} onChange={onInputChange}/>
    </form>
    
  )
}

AddCategory.propTypes={
  onNewCategory: PropTypes.func.isRequired
}