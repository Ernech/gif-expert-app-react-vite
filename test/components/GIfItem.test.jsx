import { GifItem } from "../../src/components/GifItem";
import {render, screen} from '@testing-library/react';


describe('Test at <GIfItem/>', () => {
  const url = "https://media4.giphy.com/media/2C2qwckZzyiz8UzvzK/giphy.gif?cid=f39c7d815ksqw6wil18rddwi1i867qgfzub5mxcqvysglrb7&ep=v1_gifs_search&rid=giphy.gif&ct=g"
  const title = "Dog What GIF by Best Friends Animal Society"
    
  test('should match with snapshot', () => {
        const {container} = render(<GifItem title={title} url ={url}/>)   
        expect(container).toMatchSnapshot()     
    });

  test('Should show the image with the URL and the ALT',()=>{

    render(<GifItem title={title} url={url}/>)
    const {src, alt} = screen.getByRole("img")
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('Should show the title inside the component', () => {
    render(<GifItem title={title} url={url}/>)
    expect(screen.getByText(title)).toBeTruthy()
  });

})
