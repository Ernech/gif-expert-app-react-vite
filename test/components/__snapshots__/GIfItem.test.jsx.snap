// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test at <GIfItem/> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dog What GIF by Best Friends Animal Society"
      src="https://media4.giphy.com/media/2C2qwckZzyiz8UzvzK/giphy.gif?cid=f39c7d815ksqw6wil18rddwi1i867qgfzub5mxcqvysglrb7&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Dog What GIF by Best Friends Animal Society
    </p>
  </div>
</div>
`;
